perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_def_mailTest.java

```diff
@@ -0,0 +1,55 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import com.google.common.collect.Lists;
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.Member;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests attribute module.
+ */
+public class urn_perun_member_attribute_def_def_mailTest {
+
+	private urn_perun_member_attribute_def_def_mail classInstance;
+	private PerunSessionImpl session;
+	private Attribute attributeToCheck;
+	private Member member;
+
+	@Before
+	public void setUp() throws Exception {
+		classInstance = new urn_perun_member_attribute_def_def_mail();
+		attributeToCheck = new Attribute(classInstance.getAttributeDefinition());
+	}
+
+	@Test(expected = WrongReferenceAttributeValueException.class)
+	public void testCheckNull() throws Exception {
+		System.out.println("testCheckNull()");
+		attributeToCheck.setValue(null);
+		classInstance.checkAttributeSemantics(session, member, attributeToCheck);
+	}
+
+	@Test(expected = WrongAttributeValueException.class)
+	public void testCheckEmailSyntax() throws Exception {
+		System.out.println("testCheckEmailSyntax()");
+		attributeToCheck.setValue("a/-+");
+		classInstance.checkAttributeSyntax(session, member, attributeToCheck);
+	}
+
+	@Test
+	public void testCorrect() throws Exception {
+		System.out.println("testCorrect()");
+		attributeToCheck.setValue("my@example.com");
+		classInstance.checkAttributeSemantics(session, member, attributeToCheck);
+	}
+
+	@Test
+	public void testCorrectSyntax() throws Exception {
+		System.out.println("testCorrectSyntax()");
+		attributeToCheck.setValue("my@example.com");
+		classInstance.checkAttributeSyntax(session, member, attributeToCheck);
+	}
+}
```

Comment:
Please add newlines at the end of files.

Reply:
Newlines added.